### DISCUSSION TITLE
NVidia support

### DISCUSSION THREAD

**Discussion by Promootheus** [waydroid/waydroid#1619] [Q&A] [answered]
Was curious about there being no NVidia GPU acceleration capability and it got me wondering about the technical restrictions of implementing it?

**Comment by ryzendew** [comment]
The devs seem to have a hateful bias against NVIDIA and i doubt that will be solved anytime soon. they have a ton of excuses that no longer apply and need to move on.

**Comment by mrvictory1** [comment, marked as answer]
> The devs seem to have a hateful bias against NVIDIA

Source? Nvidia is not supported because unlike any other GPU vendor, their own drivers are not included in Mesa. Android uses mesa drivers to provide GPU acceleration, Nvidia proprietary drivers are simply not a part of Mesa so they can't be used. And there *is* Nvidia support if proprietary drivers are uninstalled and nouveau is used. See this commit:
https://github.com/waydroid/waydroid/commit/404980b3af27ede6163d9fdcc9d870237492f058

**Reply by ghost** [reply]
Nvidia's drivers for linux are no longer proprietary. You're way out of date with this take.
https://developer.nvidia.com/blog/nvidia-transitions-fully-towards-open-source-gpu-kernel-modules/

In fact, all the major distributions, including Ubuntu and Fedora, have the opensource nvidia drivers in their repos.

**Reply by mrvictory1** [reply]
nvidia-open is not a Mesa driver either, my points apply to the "open source" driver as well.

**Reply by kay-lambdadelta** [reply]
> Nvidia's drivers for linux are no longer proprietary. You're way out of date with this take. https://developer.nvidia.com/blog/nvidia-transitions-fully-towards-open-source-gpu-kernel-modules/
> 
> In fact, all the major distributions, including Ubuntu and Fedora, have the opensource nvidia drivers in their repos.

They are not in mesa or compatible with it and they are not even open source, the actual driver just has been offloaded to a processor inside the gpu

**Reply by ghost** [reply]
@kay-lambdadelta Stop being so obtuse. Every other application is able to use the nvidia drivers regardless of them not being part of mesa. There's even an opensource nvidia-vaapi-driver package to allow applications like Firefox to use video hardware decoding with VA-API and still support nvidia.

Sure, they've offloaded the actual driver stuff to the hardware. Guess what? CPUs have been doing this for decades with microcode! Waydroid can support Nvidia if they wanted to. Not supporting it based on "principles" is not only inconsistent (they support x86_64 and amd64 just fine!), but also unfairly punishes people who chose to go with Nvidia Graphics at a time when Nvidia was the leader and they needed it to play gpu intensive games, and who aren't rich like a bunch of people in the tech industry and thus don't have the money to buy a new graphics card just for one obnoxious application.

**Reply by electrikjesus** [reply]
The answer to this question has already been marked. Android uses only Mesa drivers. No way to get around it. 
Thanks for playing

**Reply by kay-lambdadelta** [reply]
> @kay-lambdadelta Stop being so obtuse. Every other application is able to use the nvidia drivers regardless of them not being part of mesa. There's even an opensource nvidia-vaapi-driver package to allow applications like Firefox to use video hardware decoding with VA-API and still support nvidia.
> 
> Sure, they've offloaded the actual driver stuff to the hardware. Guess what? CPUs have been doing this for decades with microcode! Waydroid can support Nvidia if they wanted to. Not supporting it based on "principles" is not only inconsistent (they support x86_64 and amd64 just fine!), but also unfairly punishes people who chose to go with Nvidia Graphics at a time when Nvidia was the leader and they needed it to play gpu intensive games, and who aren't rich like a bunch of people in the tech industry and thus don't have the money to buy a new graphics card just for one obnoxious application.

There is no way to get around android requiring a specific graphics stack. None of this other stuff is relevant in any way really.

**Comment by zararashigal** [comment]
I don't think i need to remind linux veterans how the fglrx driver felt on ATI cards, amirite?
Maybe don't gatekeep people with the worse option but help them.
